packages/components/src/toolbar-button/index.js

```diff
@@ -12,63 +12,55 @@ import { useContext } from '@wordpress/element';
  * Internal dependencies
  */
 import Button from '../button';
+import ToolbarItem from '../toolbar-item';
 import ToolbarContext from '../toolbar-context';
-import AccessibleToolbarButtonContainer from './accessible-toolbar-button-container';
 import ToolbarButtonContainer from './toolbar-button-container';
 
 function ToolbarButton( {
 	containerClassName,
-	icon,
-	title,
-	shortcut,
-	subscript,
-	onClick,
 	className,
-	isActive,
-	isDisabled,
 	extraProps,
 	children,
+	...props
 } ) {
-	// It'll contain state if `ToolbarButton` is being used within
-	// `<Toolbar __experimentalAccessibilityLabel="label" />`
 	const accessibleToolbarState = useContext( ToolbarContext );
 
-	const button = (
-		<Button
-			icon={ icon }
-			label={ title }
-			shortcut={ shortcut }
-			data-subscript={ subscript }
-			onClick={ ( event ) => {
-				event.stopPropagation();
-				if ( onClick ) {
-					onClick( event );
-				}
-			} }
-			className={ classnames(
-				'components-toolbar__control',
-				className,
-			) }
-			isPressed={ isActive }
-			disabled={ isDisabled }
-			{ ...extraProps }
-		/>
-	);
-
-	if ( accessibleToolbarState ) {
+	if ( ! accessibleToolbarState ) {
+		// This should be deprecated when <Toolbar __experimentalAccessibilityLabel="label">
+		// becomes stable.
 		return (
-			<AccessibleToolbarButtonContainer className={ containerClassName }>
-				{ button }
-			</AccessibleToolbarButtonContainer>
+			<ToolbarButtonContainer className={ containerClassName }>
+				<Button
+					icon={ props.icon }
+					label={ props.title }
```

Comment:
Does it mean that `title` should be deprecated in favor of `label`?

Reply:
I'd say that these props should be deprecated:

- `title` (in favor of `label`)
- `isActive` (in favor of `isPressed`)
- `isDisabled` (in favor of `disabled`)

Still not sure what to do with `isDisabled` though.

This PR only changes the behavior of `ToolbarButton` when it's used inside `<Toolbar __experimentalAccessibilityLabel>`, so if we decide to deprecate those props, I guess it should be in another PR.

Reply:
I like the idea, let's go this way in the follow-up PRs.